src/components/Form.tsx

```diff
@@ -1,47 +1,108 @@
-import { Fragment } from 'react';
+import { Fragment, useState, useEffect } from 'react';
+import { JuzOptions } from '@/api/murojaah';
 import { Transition, Dialog } from '@headlessui/react';
+import Select from 'react-select';
 
 interface Props {
   showForm: boolean;
   setShowForm: Dispatch<SetStateAction<boolean>>;
 }
 
 export const Form = ({ showForm, setShowForm }: Props): JSX.Element => {
+  const [selectedOption, setSelectedOption] = useState(null);
+  const [disableSave, setDisableSave] = useState(true);
+  const [cancelText, setCancelText] = useState('Cancel');
+  const [showCancelConfirmation, setShowCancelConfirmation] = useState(false);
+
+  useEffect(() => {
+    if (showForm) {
+      resetStates();
+    }
+  }, [showForm]);
+
+  function resetStates(): void {
+    setSelectedOption(null);
+    setDisableSave(true);
+    setCancelText('Cancel');
+    setShowCancelConfirmation(false);
+  }
+
+  function setForm(option: Record<string, string>): void {
+    setSelectedOption(option);
+    setDisableSave(false);
+  }
+
   const Title = (): JSX.Element => {
     return (
       <Dialog.Title className="text-lg font-medium leading-6 text-gray-900">
-        Murojaah Juz
+        Create Juz Murojaah
       </Dialog.Title>
     );
   };
 
   const Content = (): JSX.Element => {
+    const selectStyle: StylesConfig = {
+      control: (base: CSSObjectWithLabel) => ({
+        ...base,
+        border: 0,
+        boxShadow: 'none',
+      }),
+    };
+
     return (
       <div className="mt-2">
-        <label>Select Juz</label>
-        {/* TODO Add searchable dropdown component,
-          react-select with highlight is good candidate */}
+        <div className="border border-gray-300">
+          <Select
+            defaultValue={selectedOption}
+            onChange={setForm}
+            options={JuzOptions}
+            placeholder={'Select Juz'}
+            isSearchable={false}
+            styles={selectStyle}
+          />
+        </div>
       </div>
     );
   };
 
   const Buttons = (): JSX.Element => {
+    function save(): void {
+      if (selectedOption) {
+        console.log('saved'); // TODO Implement save callback
+        closeForm();
+      }
+    }
+
+    function cancel(): void {
+      if (selectedOption && !showCancelConfirmation) {
+        setShowCancelConfirmation(true);
+        setCancelText('Confirm?');
+        return;
+      }
+      closeForm();
+    }
+
+    function closeForm(): void {
+      setShowForm(false);
+    }
+
     return (
       <div className="flex flex-row-reverse gap-2 mt-4">
         <button
           type="button"
-          className="px-6 py-2 bg-custom-teal hover:bg-teal-700 text-white rounded"
-          onClick={() => setShowForm(false)}
+          className="px-6 py-2 enabled:bg-custom-teal enabled:hover:bg-teal-700 disabled:bg-gray-400 text-white rounded"
+          onClick={() => save()}
+          disabled={disableSave}
```

Comment:
can we remove `disableSave` and just use this?
```suggestion
          disabled={!selectedOption}
```

Reply:
updated on https://github.com/RR-Ardhika/murojaah/pull/6/commits/02758916e2c35bc5d96640fa2c7e178bd353509a, thank you